connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala

```diff
@@ -23,65 +23,73 @@ import scala.collection.JavaConverters._
  */
 class UCSingleCatalog extends TableCatalog with SupportsNamespaces {
 
-  private var deltaCatalog: TableCatalog = null
+  private var delegating: TableCatalog = null
 
   override def initialize(name: String, options: CaseInsensitiveStringMap): Unit = {
     val proxy = new UCProxy()
     proxy.initialize(name, options)
-    deltaCatalog = Class.forName("org.apache.spark.sql.delta.catalog.DeltaCatalog")
-      .getDeclaredConstructor().newInstance().asInstanceOf[TableCatalog]
-    deltaCatalog.asInstanceOf[DelegatingCatalogExtension].setDelegateCatalog(proxy)
+    try {
+      if (options.containsKey("__TEST_NO_DELTA__")) {
```

Comment:
why pollute the options variable with such test-only stuff. isnt it cleaner to have a separate static variable that is completely isolated from the any of the production code paths like options?

Reply:
That static variable also needs to be in the production code so that we can access it here. I don't have a strong preference, users can set that static variable as well as this special option, if they read the source code.

Reply:
or I can make the delegating catalog class name as an option, so that we can set a different one other than `org.apache.spark.sql.delta.catalog.DeltaCatalog` in test.